crates/bevy_gltf/src/loader.rs

```diff
@@ -247,36 +247,40 @@ async fn load_gltf<'a, 'b>(
         .collect();
 
     // TODO: use the threaded impl on wasm once wasm thread pool doesn't deadlock on it
-    #[cfg(target_arch = "wasm32")]
-    for gltf_texture in gltf.textures() {
-        let (texture, label) =
-            load_texture(gltf_texture, &buffer_data, &linear_textures, &load_context).await?;
-        load_context.set_labeled_asset(&label, LoadedAsset::new(texture));
-    }
-
-    #[cfg(not(target_arch = "wasm32"))]
-    load_context
-        .task_pool()
-        .scope(|scope| {
-            gltf.textures().for_each(|gltf_texture| {
-                let linear_textures = &linear_textures;
-                let load_context: &LoadContext = load_context;
-                let buffer_data = &buffer_data;
-                scope.spawn(async move {
-                    load_texture(gltf_texture, buffer_data, linear_textures, load_context).await
+    // See https://github.com/bevyengine/bevy/issues/1924 for more details
+    // The taskpool use is also avoided when there is only one texture for performance reasons and
+    // to avoid https://github.com/bevyengine/bevy/pull/2725
```

Comment:
I think your issue and PR link are swapped?

Reply:
the issue is for the comment that already existed, on why we want to avoid threads in wasm
the pr is for the newline on why we want to avoided threaded for only one task